### PR TITLE
[2.9] add vsphere chart validation to automation framework

### PR DESCRIPTION
**Pull request by slickwarren**
add vsphere chart validation to automation framework

## Issue: 

https://github.com/rancher/qa-tasks/issues/936
https://github.com/rancher/qa-tasks/issues/872
 
## Problem

we need a way to use vsphere cloud provider in our automation 

## Solution

implement a similar method to what was used for aws cloud provider support. 
 
### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Comment by slickwarren** [comment]
note: the go.mod/sum changes will be reverted when this PR is moved from a draft state

**Review by Josh-Diamond — DISMISSED** [review]
very impressive, Caleb! 🎉

File: tests/v2/validation/provisioning/permutations/permutations.go [hunk 1]
```diff
@@ -190,12 +204,81 @@ func RunTestPermutations(s *suite.Suite, testNamePrefix string, client *rancher.
...
 			}
 		}
 	}
 }
 
+// RunPostClusterCloudProviderChecks does additinal checks on the cluster if there's a cloud provider set
+// on an active cluster.
+func RunPostClusterCloudProviderChecks(t *testing.T, client *rancher.Client, clusterType string, nodeTemplate *nodetemplates.NodeTemplate, testClusterConfig *clusters.ClusterConfig, clusterObject *steveV1.SteveAPIObject, rke1ClusterObject *management.Cluster) {
+	if strings.Contains(clusterType, clusters.RKE1ClusterType.String()) {
+
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/permutations/permutations.go:220]: supernit - please remove this extra line.

**Review by markusewalker — COMMENTED** [review]
Overall, LGTM. Just small supernits.

**Comment by slickwarren** [comment]
need to update datastoreURL after shepherd PR merges

File: tests/v2/validation/provisioning/rke2/README.md [hunk 2]
```diff
@@ -123,11 +123,24 @@ vmwarevsphereCredentials:
...
+
+Names of cloud provider options are typically controlled by rancher product. Hence the discrepancy in rke2 vs. rke1 AWS in-tree and out-of-tree options. 
+To use automation with a cloud provider, simply enter one of the following options in the `cloudProvider` field in the config. 
+
+### RKE2 Cloud Provider Options
+* aws-in-tree
+* aws
+* rancher-vsphere
+
+>>>>>>> d2d89d4ea (Add PSACT test files for Qase coverage)
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/README.md:143]: This needs to be removed.

**Review by markusewalker — COMMENTED** [review]
One small comment, rest looks good.

File: tests/v2/validation/provisioning/rke2/README.md (hunk 2 above)
Comment (slickwarren) [reply tests/v2/validation/provisioning/rke2/README.md]: thanks, addressed.

**Review by Josh-Diamond — APPROVED** [review]

**Review by markusewalker — DISMISSED** [review]

**Review by markusewalker — DISMISSED** [review]

File: tests/v2/validation/provisioning/permutations/permutations.go [hunk 3]
```diff
@@ -47,6 +58,9 @@ const (
 	nginxName            = "nginx"
 	defaultNamespace     = "default"
 
+	PollInterval = time.Duration(1 * time.Second)
```
Comment (susesgartner) [inline tests/v2/validation/provisioning/permutations/permutations.go:61]: If this isn't being exported please make this lower case.

**Review by susesgartner — APPROVED** [review]
LGTM

**Review by markusewalker — DISMISSED** [review]

**Review by susesgartner — DISMISSED** [review]
LGTM

**Review by susesgartner — DISMISSED** [review]

**Review by markusewalker — DISMISSED** [review]

**Review by susesgartner — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Pull request merged**